$Exercise\; 1$ \\
$The \;given\;optimization\;problem\;is:$
$$min\qquad 2x_1 - 5x_2 + 14x_3 + 7x_4 -11x_5 + 0.17x_6 - 10.2x_7$$
$$Subject\; to\;,\qquad
\sum_{i=1}^{7}x_i=280$$
$$x_1 - x_2\geq10$$
$$-3x_3+x_6-3x_7\leq205$$
$$20x_1 + 42x_6 +x_5\geq35$$
$$x_2 -x_3 +4x_7\geq24$$
$$x_1\geq12$$
$$x_2\geq0$$
$$x_3\geq-5$$
$$x_4\geq3$$
$$x_5\geq10$$
$$x_6\geq-4$$
$$x_7\geq0$$

$1.\; Solving \;the \;Optimization \;Problem$

In [1]:
!pip install -q pyomo
from pyomo.environ import *

     |████████████████████████████████| 8.9 MB 4.8 MB/s 
     |████████████████████████████████| 49 kB 4.2 MB/s 


In [2]:
# Creating a model
ex1 = ConcreteModel()

In [3]:
# declare decision variables
ex1.x1 = Var(domain=Reals)
ex1.x2 = Var(domain=Reals)
ex1.x3 = Var(domain=Reals)
ex1.x4 = Var(domain=Reals)
ex1.x5 = Var(domain=Reals)
ex1.x6 = Var(domain=Reals)
ex1.x7 = Var(domain=Reals)

In [4]:
# Declaring objective
ex1.obj = Objective(expr = 2*ex1.x1 - 5*ex1.x2 + 14*ex1.x3 + 7*ex1.x4 - 11*ex1.x5 + 0.17*ex1.x6 - 10.2*ex1.x7, sense=minimize)

In [5]:
# Declaring Constraints
ex1.con1 = Constraint(expr = ex1.x1 + ex1.x2 + ex1.x3 + ex1.x4 + ex1.x5 + ex1.x6 + ex1.x7 ==280)
ex1.con2 = Constraint(expr = ex1.x1 - ex1.x2 >=10)
ex1.con3 = Constraint(expr = -3*ex1.x3 + ex1.x6 -3*ex1.x7 <=205)
ex1.con4 = Constraint(expr = 20*ex1.x1 + 42*ex1.x6 + ex1.x5 >=35)
ex1.con5 = Constraint(expr = ex1.x2 -1*ex1.x3 + 4*ex1.x7 >=24)
ex1.con6 = Constraint(expr = ex1.x1 >=12)
ex1.con7 = Constraint(expr = ex1.x2 >=0)
ex1.con8 = Constraint(expr = ex1.x3 >=-5)
ex1.con9 = Constraint(expr = ex1.x4 >=3)
ex1.con10 = Constraint(expr = ex1.x5 >=10)
ex1.con11 = Constraint(expr = ex1.x6 >=-4)
ex1.con12 = Constraint(expr = ex1.x7 >=0)

In [6]:
ex1.pprint()

7 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x3 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x4 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x5 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x6 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x7 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fi

In [7]:
!apt-get install -y -qq glpk-utils

Selecting previously unselected package libsuitesparseconfig5:amd64.
(Reading database ... 160837 files and directories currently installed.)
Preparing to unpack .../libsuitesparseconfig5_1%3a5.1.2-2_amd64.deb ...
Unpacking libsuitesparseconfig5:amd64 (1:5.1.2-2) ...
Selecting previously unselected package libamd2:amd64.
Preparing to unpack .../libamd2_1%3a5.1.2-2_amd64.deb ...
Unpacking libamd2:amd64 (1:5.1.2-2) ...
Selecting previously unselected package libcolamd2:amd64.
Preparing to unpack .../libcolamd2_1%3a5.1.2-2_amd64.deb ...
Unpacking libcolamd2:amd64 (1:5.1.2-2) ...
Selecting previously unselected package libglpk40:amd64.
Preparing to unpack .../libglpk40_4.65-1_amd64.deb ...
Unpacking libglpk40:amd64 (4.65-1) ...
Selecting previously unselected package glpk-utils.
Preparing to unpack .../glpk-utils_4.65-1_amd64.deb ...
Unpacking glpk-utils (4.65-1) ...
Setting up libsuitesparseconfig5:amd64 (1:5.1.2-2) ...
Setting up libcolamd2:amd64 (1:5.1.2-2) ...
Setting up libamd2:amd64 

In [8]:
SolverFactory('glpk', executable='/usr/bin/glpsol').solve(ex1).write()

# ==========================================================
# = Solver Results                                         =
# ==========================================================
# ----------------------------------------------------------
#   Problem Information
# ----------------------------------------------------------
Problem: 
- Name: unknown
  Lower bound: -3035.88
  Upper bound: -3035.88
  Number of objectives: 1
  Number of constraints: 13
  Number of variables: 8
  Number of nonzeros: 26
  Sense: minimize
# ----------------------------------------------------------
#   Solver Information
# ----------------------------------------------------------
Solver: 
- Status: ok
  Termination condition: optimal
  Statistics: 
    Branch and bound: 
      Number of bounded subproblems: 0
      Number of created subproblems: 0
  Error rc: 0
  Time: 0.010315895080566406
# ----------------------------------------------------------
#   Solution Information
# ----------------------------

In [9]:
# display solution
print('\nObjective = ', ex1.obj())

print('\nDecision Variables')
print('x1 = ', ex1.x1())
print('x2 = ', ex1.x2())
print('x3 = ', ex1.x3())
print('x4 = ', ex1.x4())
print('x5 = ', ex1.x5())
print('x6 = ', ex1.x6())
print('x7 = ', ex1.x7())

print('\nConstraints')
print('Constraint 1  = ', ex1.con1())
print('Constraint 2  = ', ex1.con2())
print('Constraint 3  = ', ex1.con3())
print('Constraint 4  = ', ex1.con4())
print('Constraint 5  = ', ex1.con5())
print('Constraint 6  = ', ex1.con6())
print('Constraint 7  = ', ex1.con7())
print('Constraint 8  = ', ex1.con8())
print('Constraint 9  = ', ex1.con9())
print('Constraint 10  = ', ex1.con10())
print('Constraint 11  = ', ex1.con11())
print('Constraint 11  = ', ex1.con12())


Objective =  -3035.8799999999997

Decision Variables
x1 =  12.0
x2 =  0.0
x3 =  -5.0
x4 =  3.0
x5 =  269.25
x6 =  -4.0
x7 =  4.75

Constraints
Constraint 1  =  280.0
Constraint 2  =  12.0
Constraint 3  =  -3.25
Constraint 4  =  341.25
Constraint 5  =  24.0
Constraint 6  =  12.0
Constraint 7  =  0.0
Constraint 8  =  -5.0
Constraint 9  =  3.0
Constraint 10  =  269.25
Constraint 11  =  -4.0
Constraint 11  =  4.75


$2.\; Values\;of\;constraints:$
$$\sum_{i=1}^{7}x_i=280\qquad=280$$
$$x_1 - x_2=12 \qquad\geq10$$
$$-3x_3+x_6-3x_7=-3.25 \qquad\leq205$$
$$20x_1 + 42x_6 +x_5=341.25 \qquad\geq35$$
$$x_2 -x_3 +4x_7=24 \qquad\geq24$$
$$x_1=12 \qquad\geq12$$
$$x_2=0 \qquad\geq0$$
$$x_3=-5 \qquad\geq-5$$
$$x_4=3 \qquad\geq3$$
$$x_5=269.25 \qquad\geq10$$
$$x_6=-4 \qquad\geq-4$$
$$x_7=4.75 \qquad\geq0$$
$$Hence,\; all\; the\; constraints\; are\; satisfied.$$ \\
$3.\;Objective \;function \;value =-3035.88,\;x_1=12,\;x_2=0,\;x_3=-5,\;x_4=3,\;x_5=269.25,\;x_6=-4,\;x_7=4.75,\;Optimal Point=(12,\;0,\;-5,\;3,\;269.25,\;-4,\;4.75)$ \\
$4.\;For\;the \;first\; 4\;constraints,\; only\;first \;constraints\;i.e\;\sum_{i=1}^{7}x_i=280,\;satisfied\;the \;equality.$

$Exercise\; 2$
$$max\qquad 0.043x_1 + 0.027x_2 + 0.025x_3 + 0.022x_4 + 0.045x_5 - 0.0275y$$
$$Subject\; to \;,\qquad x_1 + 3x_2 + x_3 + 7x_4 +x_5 -y\leq100$$
$$x_2+2x_3+4x_4\geq60$$
$$\frac{2x_1+2x_2+x_3+x_4+5x_5}{x_1 - x_2 + x_3 -x_4 + x_5}\leq2.8$$
$$\frac{2x_1+15x_2+4x_3+3x_4-8x_5}{x_1 + x_2 + x_3 - x_5}\leq3$$
$$x_1,\;x_2,\;x_3,\;x_4,\;x_5\geq0$$
$$y\in[0,\;2.25]$$

$1.\;Rewriting\; the\; given\; problem\; as$
$$max\qquad 0.043x_1 + 0.027x_2 + 0.025x_3 + 0.022x_4 + 0.045x_5 - 0.0275y$$
$$Subject\; to \;,\qquad x_1 + 3x_2 + x_3 + 7x_4 +x_5 -y\leq100$$
$$x_2+2x_3+4x_4\geq60$$
$$-0.8x_1+4.8x_2-1.8x_3+3.8x_4+2.2x_5\geq0$$
$$-x_1+12x_2+x_3+3x_4-5x_5\leq0$$
$$x_1\geq0$$
$$x_2\geq0$$
$$x_3\geq0$$
$$x_4\geq0$$
$$x_5\geq0$$
$$y\geq0$$
$$y\leq2.25$$
$Asumptions$ \\
$x_1 - x_2 + x_3 -x_4 + x_5 <  0$ \\
$x_1 + x_2 + x_3 - x_5 > 0$ \\
$Since,\;the\;denominator\;of\;the\;following\;constraints\;\frac{2x_1+2x_2+x_3+x_4+5x_5}{x_1 - x_2 + x_3 -x_4 + x_5}\leq0,\; \frac{2x_1+15x_2+4x_3+3x_4-8x_5}{x_1 + x_2 + x_3 - x_5}\leq0 \;are\; not\; defined,\;and\;as\;they\;can\;not\;be\;0,\;\;so\;can\;we\;choose\;any\;of\;the \;following..$ \\ 
$x_1 - x_2 + x_3 -x_4 + x_5 >  0,\;x_1 + x_2 + x_3 - x_5 > 0;\;In\;this\;case\;the\;constraint\;3\;is\;not\;satisfied.$ \\
$or\;x_1 - x_2 + x_3 -x_4 + x_5 <  0,\;x_1 + x_2 + x_3 - x_5 < 0;\;In\;this\;case\;constraints\;2,\;3\;4\;are\;not\;satisfied..$ \\
$or\;x_1 - x_2 + x_3 -x_4 + x_5 >  0,\;x_1 + x_2 + x_3 - x_5 < 0;\;In\;this\;case\;the\;constraints\;2,\;3\;4\;are\;not\;satisfied.$ \\
$or\;x_1 - x_2 + x_3 -x_4 + x_5 <  0,\;x_1 + x_2 + x_3 - x_5 > 0;\;Only\;in\;this\;case,\;all\;the\;constraints\;are\;satisfied.\;$ \\

$2.\; Solving \;the \;Optimization \;Problem$

In [10]:
# Creating a model
ex2 = ConcreteModel()

In [11]:
# declare decision variables
ex2.x1 = Var(domain=Reals)
ex2.x2 = Var(domain=Reals)
ex2.x3 = Var(domain=Reals)
ex2.x4 = Var(domain=Reals)
ex2.x5 = Var(domain=Reals)
ex2.y = Var(domain=Reals)

In [12]:
# Declaring objective
ex2.obj = Objective(expr = 0.043*ex2.x1 + 0.027*ex2.x2 + 0.025*ex2.x3 + 0.022*ex2.x4 + 0.045*ex2.x5 - 0.0275*ex2.y, sense=maximize)

In [13]:
# Declaring Constraints
ex2.con1 = Constraint(expr = ex2.x1 + 3*ex2.x2 + ex2.x3 + 7*ex2.x4 + ex2.x5 - ex2.y <=100)
ex2.con2 = Constraint(expr = ex2.x2 + 2*ex2.x3 + 4*ex2.x4 >=60)
ex2.con3 = Constraint(expr = -0.8*ex2.x1 + 4.8*ex2.x2 -1.8*ex2.x3 + 3.8*ex2.x4 + 2.2*ex2.x5 >=0)
ex2.con4 = Constraint(expr = -1*ex2.x1 + 12*ex2.x2 + ex2.x3 +3*ex2.x4 -5*ex2.x5 <=0)
ex2.con5 = Constraint(expr = ex2.x1 >=0)
ex2.con6 = Constraint(expr = ex2.x2 >=0)
ex2.con7 = Constraint(expr = ex2.x3 >=0)
ex2.con8 = Constraint(expr = ex2.x4 >=0)
ex2.con9 = Constraint(expr = ex2.x5 >=0)
ex2.con10 = Constraint(expr = ex2.y >=0)
ex2.con11 = Constraint(expr = ex2.y <=2.25)

In [14]:
ex2.pprint()

6 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x3 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x4 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x5 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    y : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals

1 Objective Declarations
    obj : Size=1, Index=None, Active=True
  

In [15]:
SolverFactory('glpk', executable='/usr/bin/glpsol').solve(ex2).write()

# ==========================================================
# = Solver Results                                         =
# ==========================================================
# ----------------------------------------------------------
#   Problem Information
# ----------------------------------------------------------
Problem: 
- Name: unknown
  Lower bound: 3.939375
  Upper bound: 3.939375
  Number of objectives: 1
  Number of constraints: 12
  Number of variables: 7
  Number of nonzeros: 27
  Sense: maximize
# ----------------------------------------------------------
#   Solver Information
# ----------------------------------------------------------
Solver: 
- Status: ok
  Termination condition: optimal
  Statistics: 
    Branch and bound: 
      Number of bounded subproblems: 0
      Number of created subproblems: 0
  Error rc: 0
  Time: 0.03363656997680664
# ----------------------------------------------------------
#   Solution Information
# -----------------------------

In [16]:
# display solution
print('\nObjective = ', ex2.obj())

print('\nDecision Variables')
print('x1 = ', ex2.x1())
print('x2 = ', ex2.x2())
print('x3 = ', ex2.x3())
print('x4 = ', ex2.x4())
print('x5 = ', ex2.x5())
print('y = ', ex2.y())

print('\nConstraints')
print('Constraint 1  = ', ex2.con1())
print('Constraint 2  = ', ex2.con2())
print('Constraint 3  = ', ex2.con3())
print('Constraint 4  = ', ex2.con4())
print('Constraint 5  = ', ex2.con5())
print('Constraint 6  = ', ex2.con6())
print('Constraint 7  = ', ex2.con7())
print('Constraint 8  = ', ex2.con8())
print('Constraint 9  = ', ex2.con9())
print('Constraint 10  = ', ex2.con10())
print('Constraint 11  = ', ex2.con11())


Objective =  3.939375000000004

Decision Variables
x1 =  0.0
x2 =  0.0
x3 =  30.0
x4 =  0.0
x5 =  72.2500000000001
y =  2.25

Constraints
Constraint 1  =  100.0000000000001
Constraint 2  =  60.0
Constraint 3  =  104.95000000000024
Constraint 4  =  -331.2500000000005
Constraint 5  =  0.0
Constraint 6  =  0.0
Constraint 7  =  30.0
Constraint 8  =  0.0
Constraint 9  =  72.2500000000001
Constraint 10  =  2.25
Constraint 11  =  2.25


$3.\;Optimal\;solution\;value =3.939375000000004,\;x_1=0,\;x_2=0,\;x_3=30,\;x_4=0,\;x_5=72.2500000000001,\;y=2.25$ \\
$Activities \;of\;constraints:$
$$x_1 + 3x_2 + x_3 + 7x_4 +x_5 -y=100.0 \qquad\leq100$$
$$x_2+2x_3+4x_4=60 \qquad\geq60$$
$$-0.8x_1+4.8x_2-1.8x_3+3.8x_4+2.2x_5 =104.95000000000024 \qquad\geq0$$
$$-x_1+12x_2+x_3+3x_4-5x_5=-331.2500000000005 \qquad\leq0$$
$$x_1=0 \qquad\geq0$$
$$x_2=0 \qquad\geq0$$
$$x_3=30 \qquad\geq0$$
$$x_4=0 \qquad\geq0$$
$$x_5=72.2500000000001 \qquad\geq0$$
$$y=2.25 \qquad\geq0$$
$$y=2.25 \qquad\leq2.25$$
$4.\;Since,\;the\;denominator\;of\;the\;following\;constraints\;\frac{2x_1+2x_2+x_3+x_4+5x_5}{x_1 - x_2 + x_3 -x_4 + x_5}\leq0,\; \frac{2x_1+15x_2+4x_3+3x_4-8x_5}{x_1 + x_2 + x_3 - x_5}\leq0 \;are\; not\; defined,\;so,\;we\;can\;modify\;the\;given\;LPP\;either\;by\;adding\;two\;constraints\;as\;x_1 - x_2 + x_3 -x_4 + x_5 \neq0\;and\;x_1 + x_2 + x_3 - x_5\neq0(this\;will\;again\;require\;asumptions)\;or\;we\;can\;remove\;the\;two\;constraints\;and\;rewrite\;the\;problem\;as:$ \\
$$max\qquad 0.043x_1 + 0.027x_2 + 0.025x_3 + 0.022x_4 + 0.045x_5 - 0.0275y$$
$$Subject\; to \;,\qquad x_1 + 3x_2 + x_3 + 7x_4 +x_5 -y\leq100$$
$$x_2+2x_3+4x_4\geq60$$
$$x_1,\;x_2,\;x_3,\;x_4,\;x_5\geq0$$
$$y\in[0,\;2.25]$$


**Exercise 3**

$1.\;Let\;x_1\;and\;x_2\;be\;the\;production\;amounts\;of\;P\;and\;Q\;in\;Kg\;in\;September.\; Then, \;the\; objective\;function\;would\;be:$
 $$min\qquad\;Production\;Cost= 145x_1 + 280x_2$$
 $$Subject\;to\;,\; x_1+x_2\geq110+80+84$$
 $$\implies  x_1+x_2\geq274$$
 $$x_1\geq84;$$
 $$x_2\geq80;$$
 $$69x_1 + 55x_2\leq25800 \;(Time\;is\;converted\;into\;minutes)$$

$2.\;Solving\;the\;optimization\;problem$

In [17]:
# Creating a model
ex3 = ConcreteModel()

In [18]:
# declare decision variables
ex3.x1 = Var(domain=Reals)
ex3.x2 = Var(domain=Reals)

In [19]:
# Declaring objective
ex3.obj = Objective(expr = 145*ex3.x1 + 280*ex3.x2, sense = minimize)

In [20]:
#Declaring Constraints
ex3.con1 = Constraint(expr = ex3.x1 + ex3.x2 >=274)
ex3.con2 = Constraint(expr = ex3.x1 >=84)
ex3.con3 = Constraint(expr = ex3.x2 >=80)
ex3.con4 = Constraint(expr = 69*ex3.x1 + 55*ex3.x2 <=25800)

In [21]:
ex3.pprint()

2 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals

1 Objective Declarations
    obj : Size=1, Index=None, Active=True
        Key  : Active : Sense    : Expression
        None :   True : minimize : 145*x1 + 280*x2

4 Constraint Declarations
    con1 : Size=1, Index=None, Active=True
        Key  : Lower : Body    : Upper : Active
        None : 274.0 : x1 + x2 :  +Inf :   True
    con2 : Size=1, Index=None, Active=True
        Key  : Lower : Body : Upper : Active
        None :  84.0 :   x1 :  +Inf :   True
    con3 : Size=1, Index=None, Active=True
        Key  : Lower : Body : Upper : Active
        None :  80.0 :   x2 :  +Inf :   True
    con4 : Size=1, Index=None, Active=True
        Key  : Lower : Body          

In [22]:
SolverFactory('glpk', executable='/usr/bin/glpsol').solve(ex3).write()

# ==========================================================
# = Solver Results                                         =
# ==========================================================
# ----------------------------------------------------------
#   Problem Information
# ----------------------------------------------------------
Problem: 
- Name: unknown
  Lower bound: 50530.0
  Upper bound: 50530.0
  Number of objectives: 1
  Number of constraints: 5
  Number of variables: 3
  Number of nonzeros: 7
  Sense: minimize
# ----------------------------------------------------------
#   Solver Information
# ----------------------------------------------------------
Solver: 
- Status: ok
  Termination condition: optimal
  Statistics: 
    Branch and bound: 
      Number of bounded subproblems: 0
      Number of created subproblems: 0
  Error rc: 0
  Time: 0.023800134658813477
# ----------------------------------------------------------
#   Solution Information
# --------------------------------

In [23]:
# display solution
print('\nObjective = ', ex3.obj())

print('\nDecision Variables')
print('x1 = ', ex3.x1())
print('x2 = ', ex3.x2())

print('\nConstraints')
print('Constraint 1  = ', ex3.con1())
print('Constraint 2  = ', ex3.con2())
print('Constraint 3  = ', ex3.con3())
print('Constraint 4  = ', ex3.con4())


Objective =  50530.0

Decision Variables
x1 =  194.0
x2 =  80.0

Constraints
Constraint 1  =  274.0
Constraint 2  =  194.0
Constraint 3  =  80.0
Constraint 4  =  17786.0


$3. \;Optimal\; Solution\; Value=50530 (in\; Rs.), \;x_1=194Kgs,\;x_2=80Kgs$ \\
$Constraint Activities: $
$$x_1+x_2=274\qquad\geq274;$$
 $$x_1=194\qquad\geq84;$$
 $$x_2=80\qquad\geq80;$$
 $$69x_1 + 55x_2=17786\qquad\leq25800 \;(Time\;is\;converted\;into\;minutes)$$
 $All\;the\;constraints\;are\;satisfied\;and\;equality\;holds\;for\;the \;first\;constraint.$ \\
 $4.\; As\;the\;Maple\;Industry\;receives\;another\;order\;of\;102Kgs\;of\;product\;P\;and\;117Kgs\;of\;product\;Q,\;then\;they\;need\;to\;produce\;more\;102Kgs\;of\;product\;P\;and\;117Kgs\;of\;product\;Q.\;Wich\;will\;form\;the\;following\;LPP...$
 $$min\qquad\;Production\;Cost= 145x_1 + 280x_2$$
 $$Subject\;to\;,\; x_1+x_2\geq274+102+117$$
 $$Subject\;to\;,\; x_1+x_2\geq493$$
 $$x_1\geq84+102;$$
  $$\implies x_1\geq186;$$
 $$x_2\geq80+117;$$
 $$\implies x_2\geq197;$$
 $$69x_1 + 55x_2\leq25800 \;(Time\;is\;converted\;into\;minutes)$$
 $5.\;Solving\;the\;optimization\;problem$

In [24]:
# Creating a model 
ex3_2 = ConcreteModel()

In [25]:
# declare decision variables
ex3_2.x1 = Var(domain=Reals)
ex3_2.x2 = Var(domain=Reals)

In [26]:
# Declaring objective
ex3_2.obj = Objective(expr = 145*ex3_2.x1 + 280*ex3_2.x2, sense = minimize)

In [27]:
#Declaring Constraints
ex3_2.con1 = Constraint(expr = ex3_2.x1 + ex3_2.x2 >=493)
ex3_2.con2 = Constraint(expr = ex3_2.x1 >=186)
ex3_2.con3 = Constraint(expr = ex3_2.x2 >=197)
ex3_2.con4 = Constraint(expr = 69*ex3_2.x1 + 55*ex3_2.x2 <=25800)

In [28]:
ex3_2.pprint()

2 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals

1 Objective Declarations
    obj : Size=1, Index=None, Active=True
        Key  : Active : Sense    : Expression
        None :   True : minimize : 145*x1 + 280*x2

4 Constraint Declarations
    con1 : Size=1, Index=None, Active=True
        Key  : Lower : Body    : Upper : Active
        None : 493.0 : x1 + x2 :  +Inf :   True
    con2 : Size=1, Index=None, Active=True
        Key  : Lower : Body : Upper : Active
        None : 186.0 :   x1 :  +Inf :   True
    con3 : Size=1, Index=None, Active=True
        Key  : Lower : Body : Upper : Active
        None : 197.0 :   x2 :  +Inf :   True
    con4 : Size=1, Index=None, Active=True
        Key  : Lower : Body          

In [29]:
SolverFactory('glpk', executable='/usr/bin/glpsol').solve(ex3_2).write()

# ==========================================================
# = Solver Results                                         =
# ==========================================================
# ----------------------------------------------------------
#   Problem Information
# ----------------------------------------------------------
Problem: 
- Name: unknown
  Lower bound: -inf
  Upper bound: inf
  Number of objectives: 1
  Number of constraints: 5
  Number of variables: 3
  Number of nonzeros: 7
  Sense: minimize
# ----------------------------------------------------------
#   Solver Information
# ----------------------------------------------------------
Solver: 
- Status: ok
  Termination condition: other
  Statistics: 
    Branch and bound: 
      Number of bounded subproblems: 0
      Number of created subproblems: 0
  Error rc: 0
  Time: 0.009836196899414062


In [30]:
# display solution
#print('\nObjective = ', ex3_2.obj())

print('\nDecision Variables')
print('x1 = ', ex3_2.x1())
print('x2 = ', ex3_2.x2())

print('\nConstraints')
#print('Constraint 1  = ', ex3_2.con1())
print('Constraint 2  = ', ex3_2.con2())
print('Constraint 3  = ', ex3_2.con3())
print('Constraint 4  = ', ex3_2.con4())


Decision Variables
x1 =  None
x2 =  None

Constraints
Constraint 2  =  None
Constraint 3  =  None
ERROR: evaluating object as numeric value: x1
        (object: <class 'pyomo.core.base.var.ScalarVar'>)
    No value for uninitialized NumericValue object x1


ValueError: ignored

$6.\;The\;optimal\;solution\;value\;is\;unbounded\;and\;the\;values\;of\;the\;constraints\;are\;undetermined\;as\;the\;values\;of\;the\;variables\;are\;undetermined.\;Since\;the\;solution\;is\;not\;feasible.$
$Based\;on\;the\;result\;obtained,\;I\;would\;suggest\;to\;increase\;the\;total\;production\;hour\;per\;month.\;Then\;only\;we\;might\;be\;able\;to\;obtain\;a\;feasible\;solution.$